### PR TITLE
allow multiple auth providers with diff path

### PR DESCRIPTION
**Pull request by BojanZelic**
| Q               | A
| --------------- | ---
| Bug fix?        | yes|
| New feature?    | no|
| API breaks?     | no|
| Deprecations?   | no|
| Related tickets | |
| License         | Apache 2.0

### What's in this PR?
Allows using multiple auth mechanisms with different paths

### Why?
I ran into a bug when trying to enable multiple Kubernetes auth mechanisms:
ex:
```
      - type: kubernetes
        roles: ...
      - type: kubernetes
        path: cluster_a
        roles: ...
      - type: kubernetes
        path: cluster_b
        roles: ...
```

Previous behavior would error and require me to manually enable the auth provider
ex: 
`vault auth enable -path=cluster_a kubernetes`
`vault auth enable -path=cluster_b kubernetes`

Configuration would proceed correctly. 

### Additional context
Tested locally by adding multiple paths & ensuring that they would be purged if purgeUnmanagedConfig was enabled .

### Checklist

- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)
- [x] User guide and development docs updated (if needed)
- [x] Related Helm chart(s) updated (if needed)

### CODE REVIEW

**Comment by bonifaido** [comment]
@aabouzaid could you please take a look on this one, if it is valid? I guess it is!

**Comment by bonifaido** [comment]
@BojanZelic could you please check the build errors?

File: internal/vault/auth_methods.go [hunk 1]
```diff
@@ -75,6 +77,12 @@ func (v *vault) getUnmanagedAuthMethods(managedAuthMethods []auth) map[string]*a
 
 func (v *vault) configureAuthMethods() error {
 	managedAuths := extConfig.Auth
+	for i := range managedAuths {
```
Comment (aabouzaid) [inline internal/vault/auth_methods.go:80]: I'd suggest moving this to its own method and calling it `setDefaultValues` on the `Auth` struct.

**Comment by BojanZelic** [comment]
@bonifaido I think it should be good now whenever you get a chance.

**Comment by bonifaido** [comment]
Somehow I can't get the acceptance test to pass, the first time the kubernetes auth method is exercised in cluster it complains about missing client token (which I think means that the basic kubernetes auth method wasn't configured properly).

**Comment by bonifaido** [comment]
Yes now I see that logs in the configurer shows some errors:
```
 Error making API request.\n\nURL: POST https://vault.default:8200/v1/sys/auth\nCode: 405. Errors:\n\n* 1 error occurred:\n\t* unsupported operation\n\n","time":"2022-01-28T12:40:30Z"}
{"level":"info","msg":"Failed applying configuration file: /config/vault-configurer/vault-config.yml , sleeping for 4s before trying again","time":"2022-01-28T12:40:30Z"}
{"level":"info","msg":"applying config file : /config/vault-configurer/vault-config.yml","time":"2022-01-28T12:40:34Z"}
{"level":"info","msg":"checking if vault is sealed...","time":"2022-01-28T12:40:34Z"}
{"level":"info","msg":"vault is unsealed, configuring...","time":"2022-01-28T12:40:34Z"}
{"level":"error","msg":"error configuring vault: error configuring auth methods for vault: error configuring managed auth methods: error enabling kubernetes auth method in vault: Error making API request.\n\nURL: POST https://vault.default:8200/v1/sys/auth\nCode: 405. Errors:\n\n* 1 error occurred:\n\t* unsupported operation\n\n","time":"2022-01-28T12:40:34Z"}
```

**Comment by BojanZelic** [comment]
@bonifaido looks like I wasn't setting the defaults correctly. I pushed up the new fix & I tried it various different auth providers & settings and I believe it should be good now.

**Comment by bonifaido** [comment]
Thank you very much @BojanZelic, the build also looks green, and I did a manual testing of the simple Kubernetes auth provider that showed no errors! 👍

**Review by bonifaido — APPROVED** [review]

**Pull request merged**